In [6]:
from pathlib import Path
import pandas as pd
import numpy as np

In [9]:
def load_dataset():
    data_path = Path("../data/results/results.csv")

    # set column names
    col_names = [
                    "type",
                    "status",
                    "price",
                    "instalment",
                    "house_name",
                    "location",
                    "bedroom_count",
                    "bathroom_count",
                    "carport_count",
                    "land_area",
                    "building_area",
                ]

    # if not available
    if not data_path.is_file():
        # TODO: if the datasets is uploaded on kaggle, add the code link
        raise ReferenceError("Data kaggle is not yet available")

    return pd.read_csv(data_path, names= col_names)

In [10]:
housing = load_dataset()

In [11]:
housing.head(3)

type   status          price                    instalment  \
0  Rumah  Premier  Rp 2,1 Miliar   Cicilan: 7 Jutaan per bulan   
1  Rumah      NaN  Rp 4,1 Miliar  Cicilan: 15 Jutaan per bulan   
2  Rumah      NaN  Rp 3,3 Miliar  Cicilan: 12 Jutaan per bulan   

                                          house_name        location  \
0  Rumah 2 Lantai Bagus Sertifikat Hak Milik di M...  Andir, Bandung   
1                    Rumah Cantik @ Andir (Sudirman)  Andir, Bandung   
2  Jual Cepat Rumah di Maleber Rajawali Bisa Untu...  Andir, Bandung   

   bedroom_count  bathroom_count  carport_count land_area building_area  
0              3               2              2    137 m²        170 m²  
1              3               2              3    202 m²        300 m²  
2              5               2              1    350 m²        258 m²

In [12]:
housing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7611 entries, 0 to 7610
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   type            7611 non-null   object
 1   status          145 non-null    object
 2   price           7611 non-null   object
 3   instalment      7611 non-null   object
 4   house_name      7611 non-null   object
 5   location        7611 non-null   object
 6   bedroom_count   7611 non-null   int64 
 7   bathroom_count  7611 non-null   int64 
 8   carport_count   7611 non-null   int64 
 9   land_area       7609 non-null   object
 10  building_area   7610 non-null   object
dtypes: int64(3), object(8)
memory usage: 654.2+ KB


This analysis examines a DataFrame containing 7611 rows and 11 columns related to house listings. Here's a breakdown of the key points:

**Data Types:**

- Most columns (type, status, price, instalment, house_name, location) are currently stored as text (`object`) even though they might contain numerical information ("20 million").
- Numerical features (bedroom_count, bathroom_count, carport_count) have the correct data type (`int64`).
- Land and building area are numerical values stored as text (`object`) due to scraping capturing extra characters (e.g., "m²").

**Missing Values:**

- There are missing values in some columns:
    - `status` (145 missing)
    - `land_area` (2 missing)
    - `building_area` (1 missing)

**Data Cleaning Required:**

- Transform price, land_area, and building_area from text to numbers by removing characters like "Rp" and "m²".
- Decide how to handle missing values:
    - Impute missing values with appropriate strategies (e.g., median, mean) if the number of missing values is low.
    - Remove entries with missing values if they are a significant portion of the data or if imputation is not feasible.

**Additional Considerations:**

- Explore the meaning of columns like `type` and `status` and whether they require further cleaning or encoding.
- Analyze potential outliers in numerical features after cleaning.
